### DISCUSSION TITLE
Code explanation

### DISCUSSION THREAD

**Discussion by MeherabTheLord** [community/community#48724] [Programming Help]
Hi everyone, I would like to get some advise through javascript cookies handling.
I have these code for extracting values from cookies.However i understand all the way till .map from .reduce array method i have some doubt. Can any body tell me what is happening through these method. Especially in              .reduce((arg,[key,val])=>
	({...arg,[key.trim()]:decodeURIComponent(val)}),
	{});
this line.
codes:

		let cookieString="firstName=john;   lastName=Doe;password    =1234";
		const getCookies1=cookieString
				.split(";")
				.map(split=>split.split("="))
				.reduce((arg,[key,val])=>
					({...arg,[key.trim()]:decodeURIComponent(val)}),
					{});

		document.write(cookieString + `<br>`);
		document.write(getCookies1.password);

**Comment by sergeyleschev** [comment]
The reduce() method is being used to transform an array of cookie key-value pairs into a single object that contains all the keys and values of the cookies.

For each item in the array, the reduce() method takes two parameters: the accumulated value (which starts as an empty object {}) and the current item in the array (an array with two elements, key and val).

The code inside the reduce() method then spreads the accumulated value into a new object literal ({...arg}) and adds a new property with the trimmed key and decoded val. This new object is then returned and becomes the new accumulated value for the next iteration.

So basically, the .map() method splits the cookie string into an array of key-value pairs, then the .reduce() method uses this array to build an object containing all the keys and their corresponding values of the cookies.
